app/src/main/java/com/github/polybooks/database/OLBookDatabase.kt

```diff
@@ -44,6 +45,10 @@ class OLBookDatabase(private val url2json : (String) -> CompletableFuture<JsonEl
 
     override fun queryBooks(): BookQuery = OLBookQuery()
 
+    override fun addBook(book: Book): CompletableFuture<Unit> {
+        throw UnsupportedOperationException("The 'add' operation is not supported by open library database")
```

Comment:
This line can be tested for sure

Reply:
the better solution is to make so that OLBookDatabase is no longer a BookDatabase, and therefore doesn't need that method.